pkg/ingest/ingest.go

```diff
@@ -0,0 +1,101 @@
+package ingest
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	"github.com/thanos-community/obslytics/pkg/dataframe"
+	"github.com/thanos-community/obslytics/pkg/output"
+)
+
+type Ingestor interface {
+	Ingest(*storepb.Series) error
+}
+
+// Options for a single aggregation.
+type AggrOption struct {
+	// Should the aggregation be used?
+	Enabled bool
+	// Column to store the aggregation at
+	Column string
+}
+
+// Collections of aggregations-related options. Determines what aggregations are enabled etc..
+type aggrsOptions struct {
+	// If true (default), the samples should repeat the same time for every day.
+	// This puts additional requirement for 24h to be divisible by the resolution.
+	MidnightAligned bool
+
+	Sum   AggrOption
+	Count AggrOption
+	Min   AggrOption
+	Max   AggrOption
+}
+
+// By default, all aggregations are disabled and target columns set with `_` prefix.
+func NewAggregationsOptions() (ao aggrsOptions) {
+	ao.MidnightAligned = true
+
+	ao.Sum.Column = "_sum"
+	ao.Count.Column = "_count"
+	ao.Min.Column = "_min"
+	ao.Max.Column = "_max"
+	return ao
+}
+
+// Ingests the data and produces aggregations at the specified resolution.
+type AggregatedIngestor struct {
+	Resolution  time.Duration
+	aggrOptions aggrsOptions
+}
+
+// Ingest the data to an aggregated set.
+func (i AggregatedIngestor) Ingest(s *storepb.Series) error {
```

Comment:
What about just `Aggregate` ;p

Reply:
Also this tells me that this aggregator is really `StoreSereiesAggregator` right?.

Reply:
I can imagine different processing that would lets say finding continuous time blocks for the same labels/values: not sure if that would still be called aggregation. Ingest feels more neutral.

But for the specific implementation, `Aggregator` would work just fine.

Regarding possible different inputs, I wonder if there should be some interface instead to put around the different implementations, that could then be passed to the aggregators, so that we don't have to write two different ones for different inputs. What do you think? Anyway, would postpone that until we have 2 different input types.

Reply:
I've defined `input.Series` interface to abstract different input classes and reduce it only to methods we really need. The `Aggregator` now should be input independent.